game/src/edit/roads.rs

```diff
@@ -508,8 +508,34 @@ fn can_reverse(_: LaneType) -> bool {
     lt == LaneType::Driving || lt == LaneType::Biking || lt == LaneType::Bus
 }*/
 
+fn default_outside_lane_placement(road: &mut EditRoad, dir: Direction) -> usize {
+    let idx = if road.lanes_ltr.first().unwrap().dir == dir {
+        if road.lanes_ltr.first().unwrap().lt.is_walkable() {
+            1
+        } else {
+            0
+        }
+    } else {
+        if road.lanes_ltr.last().unwrap().lt.is_walkable() {
+            road.lanes_ltr.len() - 1
+        } else {
+            road.lanes_ltr.len()
+        }
+    };
+    idx
+}
+
 // Returns the index where the new lane was inserted
 fn add_new_lane(road: &mut EditRoad, lt: LaneType) -> usize {
+
+    // Default to Fwd as long as the lane directions are equal or there are more with Back
+    // if there are more lanes with Fwd then use Back
+    let dir = if (road.lanes_ltr.iter().filter(|x| x.dir == Direction::Fwd).count() as f64 / road.lanes_ltr.len() as f64) <= 0.5 {
```

Comment:
You can auto-format the code and get better line-wrapping here by running `cargo +nightly fmt`. You have to install the nightly toolchain for the formatting, unfortunately, because we [use a few overrides](https://github.com/a-b-street/abstreet/blob/master/rustfmt.toml) that aren't stable yet.

Reply:
Very cool I will check it out